### ISSUE TITLE
support reduction of information arrays w/ time dimension

### ISSUE THREAD

**Issue by taobrienlbl** [LBL-EESA/TECA#389]
The `teca_temporal_reduction` code does not support reducing information arrays. This would be useful for reducing time-dependent information arrays in the bayesian AR detector: e.g., I was just trying to average the `ar_count` variable.  This will be relevant for #369 too, since this would allow a temporal reduction to be incorporated into a variant of the bayesian AR detector app.

**Comment by burlen** [comment]
good point, I'd like to incorporate this and other improvement to the C++ port. This would occur over an intermediate time period rather than right away. Is this OK? If not then we can add to the Python code now.
